lib/exception/network_exceptions.dart

```diff
@@ -0,0 +1,159 @@
+              break;
+            case DioErrorType.RECEIVE_TIMEOUT:
+              networkExceptions = NetworkExceptions.sendTimeout();
+              break;
+            case DioErrorType.RESPONSE:
+              switch (error.response.statusCode) {
+                case 400:
+                  networkExceptions = NetworkExceptions.unauthorisedRequest();
+                  break;
+                case 401:
+                  networkExceptions = NetworkExceptions.unauthorisedRequest();
+                  break;
+                case 403:
+                  networkExceptions = NetworkExceptions.unauthorisedRequest();
+                  break;
+                case 404:
+                  networkExceptions = NetworkExceptions.notFound("Not found");
+                  break;
+                case 409:
+                  networkExceptions = NetworkExceptions.conflict();
+                  break;
+                case 408:
+                  networkExceptions = NetworkExceptions.requestTimeout();
+                  break;
+                case 500:
+                  networkExceptions = NetworkExceptions.internalServerError();
+                  break;
+                case 503:
+                  networkExceptions = NetworkExceptions.serviceUnavailable();
+                  break;
+                default:
+                  var responseCode = error.response.statusCode;
+                  networkExceptions = NetworkExceptions.defaultError(
+                    "Received invalid status code: $responseCode",
+                  );
+              }
+              break;
+            case DioErrorType.SEND_TIMEOUT:
+              networkExceptions = NetworkExceptions.sendTimeout();
+              break;
+          }
+        } else if (error is SocketException) {
+          networkExceptions = NetworkExceptions.noInternetConnection();
+        } else {
+          networkExceptions = NetworkExceptions.unexpectedError();
+        }
+        return networkExceptions;
+      } on FormatException catch (_) {
+        // Helper.printError(e.toString());
+        return NetworkExceptions.formatException();
+      } catch (_) {
+        return NetworkExceptions.unexpectedError();
+      }
+    } else {
+      if (error.toString().contains("is not a subtype of")) {
+        return NetworkExceptions.unableToProcess();
+      } else {
+        return NetworkExceptions.unexpectedError();
+      }
+    }
+  }
+
+  static String getErrorMessage(NetworkExceptions networkExceptions) {
+    var errorMessage = "";
+    networkExceptions.when(notImplemented: () {
+      errorMessage = "Not Implemented";
+    }, requestCancelled: () {
+      errorMessage = "Request Cancelled";
+    }, internalServerError: () {
+      errorMessage = "Internal Server Error";
+    }, notFound: (String reason) {
+      errorMessage = reason;
+    }, serviceUnavailable: () {
+      errorMessage = "Service unavailable";
+    }, methodNotAllowed: () {
+      errorMessage = "Method Allowed";
```

Comment:
@sleepylee this one should be `Method Not Allowed` 😄

Reply:
ohh 🤔  will fix it